src/map/entities/charentity.cpp

```diff
@@ -2134,7 +2134,16 @@ void CCharEntity::OnRaise()
             // Player died within a battlefield, reward the battlefield level equivalent EXP
             if (StatusEffectContainer->HasStatusEffect(EFFECT_BATTLEFIELD))
             {
-                expLost = m_LevelRestriction >= 1 && m_LevelRestriction <= 67 ? (charutils::GetExpNEXTLevel(m_LevelRestriction) * 8) / 100 : 2400;
+                if (PBattlefield != nullptr && (PBattlefield->GetRuleMask() & RULES_LOSE_EXP) != RULES_LOSE_EXP)
+                {
+                    expLost = 0;
+                }
+                else
+                {
+                    expLost = m_LevelRestriction >= 1 && m_LevelRestriction <= 67
+                                  ? (charutils::GetExpNEXTLevel(m_LevelRestriction) * 8) / 100
+                                  : 2400;
```

Comment:
Were you getting sanity errors with this all on one line?

Reply:
Yeah I was. The way it is currently was actually what the sanity check suggested I do.

Reply:
Grrooooooss